pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -364,6 +364,9 @@ func (c *Controller) syncService(key string) error {
 		return err
 	}
 
+	// Drop EndpointSlices that have been marked for deletion to prevent the controller from getting stuck.
+	endpointSlices = dropEndpointSlicesPendingDeletion(endpointSlices)
+
 	if c.endpointSliceTracker.StaleSlices(service, endpointSlices) {
```

Comment:
@robscott I can't remember all the details now of the tracker, is there anything else we should do in the `endpointSliceTracker`?

Reply:
Commenting for future us - this is right, explained by https://github.com/kubernetes/kubernetes/pull/109624/files#r860592618.